### ISSUE TITLE
One of the Ancient Data Caches signalogic broken

### ISSUE THREAD

**Issue by AlexS-empyrion** [RavienGaming/Reforged-Eden-2#326]
### Reforged Eden 2 Build Number

22

### Game mode

Survival

### Tested in vanilla

Not tested in vanilla or not applicable

### Description of issue

One of the Ancient Data Caches have broken signalogic. After Completing hack blast doors and LCDs are not trigger.
Name is: RE2_BAO_ProgHackT1_01
![RE2-1 11-Omicron_2024-09-26_10-52-41](https://github.com/user-attachments/assets/e98e07b3-76b3-464f-93f1-7e4ca41a5899)

### POI Name

RE2_BAO_ProgHackT1_01

**Comment by RavienGaming** [comment]
This is likely a game bug and will need to be reproduced and reported using a vanilla save game.
When reproducing a game bug in a vanilla save, make sure you are using the same mode _(singleplayer/co-op/dedicated multiplayer, survival/creative)_ as what you were using when the issue happened in the scenario.
To see a list of known game bugs or to report a new one please see: https://empyriononline.com/forums/bugs.24/

**Comment by AlexS-empyrion** [comment]
I don't think it's game bug because problem is only with this one. Others are work as intended.
It might be because this one have success switch not active.

**Comment by AlexS-empyrion** [comment]
Never mind. I've checked. There is only one version in library. Switch is active there. So it is possible that dialog instead of trigger signal turns off switch. It's hard to reproduce bug even within scenario. And I don't see how it can be reproduced in vanilla. Because you will need your own dialog that triggers signal

**Comment by AlexS-empyrion** [comment]
And I found this messages in game log:
26-07:47:27.442 10_47 -WRN- Dialog: Cannot determine state of signal 'Lockdown'
26-07:47:27.443 10_47 -WRN- Dialog: Cannot determine state of signal 'Success'
26-07:48:02.756 10_48 -ERR- Dialog: Cannot set state of signal 'Success'

**Comment by AlexS-empyrion** [comment]
Can problem be caused you not near enough to POI for game to read signals from it?

**Comment by RavienGaming** [comment]
> Can problem be caused you not near enough to POI for game to read signals from it?

No, this is a game bug with general signal logic issues on the dialogue. I need to somehow find the time to report it.

**Comment by StHotPants** [comment]
![image](https://github.com/user-attachments/assets/88113cd2-e091-42e1-ae3e-6976028953da)
same issue with the same cache version both in MP and SP, can solve the "guess the number" but the cache never opens due to the signal logic issue

**Comment by Powermonkey00** [comment]
Can verify, I've had the same issue with the same style data cache on a MP server on build 23.  I found a cluster of these at an exploration site, and none of the caches in this style would open.  The blocks that are supposed to be destroyed become 'steel block (Signal) - Destroyed' with full health
![image](https://github.com/user-attachments/assets/b265c8ad-13ac-4199-a72c-a3431088c7e7)

**Comment by RavienGaming** [comment]
Yes until Eleon fixes the bug this will continue to happen. I may need to redo the hacking so that it simply gives rewards via dialogue for now until I have time to report the bug.

**Comment by AlexS-empyrion** [comment]
I make some testing. Problem occurs only when I staying on outside of console. Seems dialog does not register POI and doesn't see signals when I staying outside.
Workaround is make sure you start dialog when stay left or right side of console. It this case game registers that you in POI.

**Comment by RavienGaming** [comment]
> I make some testing. Problem occurs only when I staying on outside of console. Seems dialog does not register POI and doesn't see signals when I staying outside. Workaround is make sure you start dialog when stay left or right side of console. It this case game registers that you in POI.

Oh that is interesting. Thanks.

**Comment by byo13** [comment]
Just happened to me.
I was a little far from the console, confirming what AlexS-empyrion suggested.
Maybe making the console inside some strict area, like when we're hacking satellites, could prevent this from happening?
Thanks, guys!

**Comment by RavienGaming** [comment]
I'll have to report this as a game bug to the developers to see if it's anything they can fix on their end, but yeah in the meantime I'll see if I can figure something out.

**Comment by AlexS-empyrion** [comment]
You can move console far inside the structure to be accessible only within any structure block.

**Comment by RavienGaming** [comment]
> You can move console far inside the structure to be accessible only within any structure block.

Yeah, but that would require a redesign of every hacking POI and it still doesn't actually fix the problem as you'll encounter the same issue in other places as well.

**Comment by AlexS-empyrion** [comment]
> Yeah, but that would require a redesign of every hacking POI and it still doesn't actually fix the problem as you'll encounter the same issue in other places as well.

Not everyone but only this one. Others doesn't have this issue.
Ex. 4 switch Data Cache have console within a structure.  Other ones don't use signals that transfers to POI.

**Comment by RavienGaming** [comment]
This should be fixed in the next update via a check I am adding in the dialogue. If it cannot detect the structure it will warn you to move closer to the POI or stand on part of it before it will let you initiate the hack. This should fix the issue from happening.
Thanks to RexXxuS for the tip on this.

**Comment by dot-sent** [comment]
@RavienGaming The same issue just happened to me on the same POI in build 42. 

Messages in the log:
```
12-08:34:06.862 09_34 -WRN- Dialog: Cannot determine state of signal 'Lockdown'
12-08:34:06.862 09_34 -WRN- Dialog: Cannot determine state of signal 'Success'
12-08:34:52.322 09_34 -ERR- Cannot change unknown signal: Success
12-08:35:21.942 09_35 -WRN- Dialog: Cannot determine state of signal 'Lockdown'
```

My character was indeed floating on the outer side of the POI and I was communicating with the console probably from around the farthest distance possible.

Not a big deal in SP (I got the reward via gm'ing inside), but it would be nice to have this fixed once and for all.

**Comment by RavienGaming** [comment]
> [@RavienGaming](https://github.com/RavienGaming) The same issue just happened to me on the same POI in build 42.
> 
> Messages in the log:
> 
> ```
> 12-08:34:06.862 09_34 -WRN- Dialog: Cannot determine state of signal 'Lockdown'
> 12-08:34:06.862 09_34 -WRN- Dialog: Cannot determine state of signal 'Success'
> 12-08:34:52.322 09_34 -ERR- Cannot change unknown signal: Success
> 12-08:35:21.942 09_35 -WRN- Dialog: Cannot determine state of signal 'Lockdown'
> ```
> 
> My character was indeed floating on the outer side of the POI and I was communicating with the console probably from around the farthest distance possible.
> 
> Not a big deal in SP (I got the reward via gm'ing inside), but it would be nice to have this fixed once and for all.

This should be fixed via a check that prevents you from using it if you interact with the console and it cannot detect the core.
Can you use the **di** console command, look at the POI, and share a screenshot please so I can get the blueprint name?

**Comment by dot-sent** [comment]
> 
> This should be fixed via a check that prevents you from using it if you interact with the console and it cannot detect the core. Can you use the **di** console command, look at the POI, and share a screenshot please so I can get the blueprint name?

<img width="1920" height="1080" alt="Image" src="https://github.com/user-attachments/assets/e0259c83-6532-403f-b6f4-9bd54ba54ef1" />

Here it is. I took your request as a mission and jumped back over 7 hops to get it :D 

I have also found another instance of the same POI in a different system and experimented a bit with it. I couldn't reproduce the issue any longer - if my character is further away from the POI, interacting with the console indeed gives a warning to move closer.

I assume the earlier bug might've been some super-edge case, I vaguely remember that my character was actually still moving in space when I initiated the dialog.

**Comment by dot-sent** [comment]
@RavienGaming I accidentally got this issue reproduced again and figured out both a reproducible case and a workaround. The issue is 100% reproducible _if during the dialog one of my ships is closer to my character than the POI_. So, if I fly to the POI so that the nose of my ship almost touches the console and _then_ do the hacking, the issue is reproduced 100%. The same happens if I drive my SV deeper into the POI that has the console further inside.

**Comment by RavienGaming** [comment]
> [@RavienGaming](https://github.com/RavienGaming) I accidentally got this issue reproduced again and figured out both a reproducible case and a workaround. The issue is 100% reproducible _if during the dialog one of my ships is closer to my character than the POI_. So, if I fly to the POI so that the nose of my ship almost touches the console and _then_ do the hacking, the issue is reproduced 100%. The same happens if I drive my SV deeper into the POI that has the console further inside.

Ah yes, that would happen due to you being inside your ship, not the POI, so any signals are sent to your ship. I could potentially add a check to make sure it's actually a POI, but that still wouldn't stop every case and I don't think it's worth it because there are other issues that can cause this too.
